### PR TITLE
python310Packages.nc_dnsapi: init at 0.1.5

### PR DESCRIPTION
**Pull request by veehaitch**
###### Description of changes

Add Python package `nc_dnsapi`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Trundle — APPROVED** [review]

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6150,6 +6150,8 @@ in {
 
   nbxmpp = callPackage ../development/python-modules/nbxmpp { };
 
+  nc_dnsapi = callPackage ../development/python-modules/nc_dnsapi { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6153]:
```suggestion
  nc-dnsapi = callPackage ../development/python-modules/nc-dnsapi { };
```

File: pkgs/development/python-modules/nc_dnsapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "nc_dnsapi";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nc_dnsapi/default.nix:8]:
```suggestion
  pname = "nc-dnsapi";
```

File: pkgs/development/python-modules/nc_dnsapi/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.1.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-1fvzr3e0ZAbSDOovhLz5GHJCS6l+K89fbYHoaWxO9cA=";
+  };
+
+  propagatedBuildInputs = [ requests ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nc_dnsapi/default.nix:17]:
```suggestion

  # no tests
  doCheck = false;

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
package names must be normalized

**Pull request merged**